wallet/wallet.go

```diff
@@ -3161,6 +3161,46 @@ func (w *Wallet) ImportScript(rs []byte) error {
 	return nil
 }
 
+// RemoveScript removes the provided script from the wallet if it exists.
+func (w *Wallet) RemoveScript(script []byte) error {
```

Comment:
A better API here would be to pass the address itself instead of a script.  It should only succeed if the address being removed is imported, and if it is P2SH, then the script should be removed as well.  This would allow the same API to also be used to remove addresses for imported pubkeys or privkeys.